lib/rules.js

```diff
@@ -353,6 +353,338 @@ export const GROUP_TERMS_BY_ROOT = defineRule(
+
+                // Theoretical second term (should be 2 * 2 * 3 * x -> 12x in e.g.)
+                const secondCoef = build.numberNode(2 * nthRoot(firstCoef * thirdCoef, 2))
+                const firstMiddle = firstDegree / 2 == 1
+                      ? firstPoly.args[0]
+                      : build.applyNode('pow', [firstPoly.args[0], build.numberNode(firstDegree / 2)])
+
+                let secondMiddle
+                if (thirdPoly) {
+                    secondMiddle = thirdDegree / 2 == 1
+                        ? thirdPoly.args[0]
+                        : build.applyNode('pow', [thirdPoly.args[0], build.numberNode(thirdDegree / 2)])
+                }
+
+                const secondPoly = secondMiddle
+                      ? build.applyNode('mul', [firstMiddle, secondMiddle], {implicit: true})
+                      : firstMiddle
+
+                const matchSecondTerm =  build.applyNode ('mul',
+                                                     [secondCoef, secondPoly]
+                                                          , {implicit: true})
+
+                // 1. Checks if coefficients are perfect squares
+                // 2. Checks if leading degree is even
+                // 3. Middle coefficient follows condition
+                isPerfectSquareTrinomial = Number.isInteger(nthRoot(firstCoef, 2))
+                    && Number.isInteger(nthRoot(thirdCoef, 2))
+                    && firstDegree % 2 == 0
+                    && thirdDegree ? thirdDegree % 2 == 0 : true
+                    && (print(secondTerm) == print(flattenOperands(matchSecondTerm))
+                        || print(secondTerm) == '-' + print(flattenOperands(matchSecondTerm)))
+            }
+        }
+        return isPerfectSquareTrinomial ? {node} : null
+    },
+
+    (node) => {
+        const [firstTerm, secondTerm, thirdTerm] = node.args
+        const firstCoef = query.getValue(firstTerm.args[0])
+        const firstPoly = firstTerm.args[1]
+        const firstDegree = query.getValue(firstPoly.args[1])       
+        const thirdCoef = isPolynomialTerm(thirdTerm) && !query.isNumber(thirdTerm)
+              ? query.getValue(thirdTerm.args[0])
+              : query.getValue(thirdTerm)
+        const thirdPoly = isPolynomialTerm(thirdTerm) && !query.isNumber(thirdTerm)
+              ? thirdTerm.args[1]
+              : null
+        const thirdDegree = thirdPoly ? query.getValue(thirdPoly.args[1]) : null
+
+        const firstFactor = parse(`${Math.sqrt(firstCoef)} ${getVariableFactorName(firstPoly)}^${firstDegree/2}`)
+        const secondFactor = thirdPoly
+              ? `${Math.sqrt(thirdCoef)} ${getVariableFactorName(thirdPoly)}^${thirdDegree/2}`
+              : Math.sqrt(query.getValue(thirdTerm))
+
+        const result = query.isNeg(secondTerm)
+              ? parse(`(${print(firstFactor)} - ${secondFactor})^2`)
+              : parse(`(${print(firstFactor)} + ${secondFactor})^2`)
+        return result
```

Comment:
I've update `math-nodes` to have more build functions, see https://github.com/semantic-math/math-nodes/commit/9dbd3e71bef309b0e406b19a5c152d5f1952fd2a.

This would become:
```
const result = query.isNeg(secondTerm)
   ? build.add(clone(firstFactor), clone(secondFactor))
   : build.sub(clone(firstFactor), clone(secondFactor))
```